src/Native/Unix/System.Security.Cryptography.Native/pal_x509.c

```diff
@@ -259,6 +304,26 @@ X509VerifyStatusCode CryptoNative_X509StoreCtxGetError(X509_STORE_CTX* ctx)
     return (unsigned int)X509_STORE_CTX_get_error(ctx);
 }
 
+int32_t CryptoNative_X509StoreCtxReset(X509_STORE_CTX* ctx)
+{
+    X509* leaf = X509_STORE_CTX_get0_cert(ctx);
+    X509Stack* untrusted = X509_STORE_CTX_get0_untrusted(ctx);
+    X509_STORE* store = X509_STORE_CTX_get0_store(ctx);
```

Comment:
It seems like this change broke our RedHat6 official build:

```
/__w/1/s/src/Native/Unix/System.Security.Cryptography.Native/pal_x509.c(311,25): error GC97D5F25: implicit declaration of function 'X509_STORE_CTX_get0_store' is invalid in C99 [-Werror,-Wimplicit-function-declaration] [/__w/1/s/src/Native/build-native.proj]
      X509_STORE* store = X509_STORE_CTX_get0_store(ctx);
                          ^
/__w/1/s/src/Native/Unix/System.Security.Cryptography.Native/pal_x509.c(311,17): error G5F2A7227: incompatible integer to pointer conversion initializing 'X509_STORE *' (aka 'struct x509_store_st *') with an expression of type 'int' [-Werror,-Wint-conversion] [/__w/1/s/src/Native/build-native.proj]
      X509_STORE* store = X509_STORE_CTX_get0_store(ctx);
                  ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/__w/1/s/src/Native/Unix/System.Security.Cryptography.Native/pal_x509.c(758,21): error G5F2A7227: incompatible integer to pointer conversion initializing 'X509_STORE *' (aka 'struct x509_store_st *') with an expression of type 'int' [-Werror,-Wint-conversion] [/__w/1/s/src/Native/build-native.proj]
          X509_STORE* store = X509_STORE_CTX_get0_store(storeCtx);
                      ^       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  3 errors generated.
  make[2]: *** [System.Security.Cryptography.Native/CMakeFiles/objlib.dir/pal_x509.c.o] Error 1
  make[1]: *** [System.Security.Cryptography.Native/CMakeFiles/objlib.dir/all] Error 2
  make: *** [all] Error 2
  Failed to build corefx native components.
/__w/1/s/src/Native/build-native.proj(37,5): error MSB3073: The command ""/__w/1/s/src/Native/build-native.sh" x64 Release Linux outconfig netcoreapp-Linux-Release-x64 stripsymbols" exited with code 1.

Build FAILED.

/__w/1/s/src/Native/Unix/System.Security.Cryptography.Native/pal_x509.c(311,25): error GC97D5F25: implicit declaration of function 'X509_STORE_CTX_get0_store' is invalid in C99 [-Werror,-Wimplicit-function-declaration] [/__w/1/s/src/Native/build-native.proj]
/__w/1/s/src/Native/Unix/System.Security.Cryptography.Native/pal_x509.c(311,17): error G5F2A7227: incompatible integer to pointer conversion initializing 'X509_STORE *' (aka 'struct x509_store_st *') with an expression of type 'int' [-Werror,-Wint-conversion] [/__w/1/s/src/Native/build-native.proj]
/__w/1/s/src/Native/Unix/System.Security.Cryptography.Native/pal_x509.c(758,21): error G5F2A7227: incompatible integer to pointer conversion initializing 'X509_STORE *' (aka 'struct x509_store_st *') with an expression of type 'int' [-Werror,-Wint-conversion] [/__w/1/s/src/Native/build-native.proj]
/__w/1/s/src/Native/build-native.proj(37,5): error MSB3073: The command ""/__w/1/s/src/Native/build-native.sh" x64 Release Linux outconfig netcoreapp-Linux-Release-x64 stripsymbols" exited with code 1.
    0 Warning(s)
    4 Error(s)
```

https://dev.azure.com/dnceng/internal/_build/results?buildId=104405&view=logs&jobId=7cbf968d-26f0-55c8-fa2c-bd900b311c34&taskId=80b36c5e-5cc3-5f6d-94ed-94388cf0917b&lineStart=1348&lineEnd=1348&colStart=1&colEnd=14